core/cmd/client.go

```diff
@@ -229,23 +237,32 @@ func (n ChainlinkAppFactory) NewApplication(ctx context.Context, cfg chainlink.G
 		UnrestrictedHTTPClient:   unrestrictedClient,
 		SecretGenerator:          chainlink.FilePersistedSecretGenerator{},
 		LoopRegistry:             loopRegistry,
+		Telemetry:                telem,
 	})
 }
 
-func setupSolanaRelayer(appLggr logger.Logger, db *sqlx.DB, cfg chainlink.GeneralConfig, loopRegistry *plugins.LoopRegistry, ks keystore.Solana) (loop.Relayer, error) {
+type relayerFactory struct {
```

Comment:
nice. i tried to figure out a way to reduce the copy and paste in my PR, but a factory with `NewSolana` and `NewStarknet` didn't occur to me. much better.